koans/src/java8/AboutLambdas.java

```diff
@@ -0,0 +1,91 @@
+package java8;
+
+import com.sandwich.koan.Koan;
+
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import static com.sandwich.util.Assert.assertEquals;
+import static com.sandwich.koan.constant.KoanConstants.__;
+
+interface Caps {
+    public String capitalize(String name);
+}
+
+public class AboutLambdas {
+
+    String fieldFoo = "Lambdas";
+
+    @Override
+    public String toString() {
+        return "CAPS";
+    }
+
+    static volatile String str = "";
+
+    //lambda has access to "this"
+    Caps lambdaField = s -> this.toString();
+    //lambda has access to object methods
+    Caps lambdaField2 = s -> toString();
+
+    @Koan
+    public void verySimpleLambda() throws InterruptedException {
+        Runnable r8 = () -> str = "from other thread";
+        r8.run();
```

Comment:
You are right, I guess I better spawn another thread. (Most of literature is using Runnable to demonstrate simple lambda form, so I thought it would be a good idea to have such example).